lua/nvim-treesitter/configs.lua

```diff
@@ -295,24 +299,17 @@ end
 -- Setup call for users to override module configurations.
 -- @param user_data module overrides
 function M.setup(user_data)
-  for name, data in pairs(user_data) do
-    if name == 'ensure_installed' then
-      config.ensure_installed = data
-      require'nvim-treesitter.install'.ensure_installed(data)
-    else
-      config.modules[name] = vim.tbl_deep_extend('force', config.modules[name] or {}, data)
-
-      recurse_modules(function(_, _, new_path)
-        if data.enable then
-          enable_all(new_path)
-        end
-
-        for _, lang in ipairs(data.disable or {}) do
-          disable_mod_conf_autocmd(new_path, lang)
-        end
-      end, config.modules)
+  config.modules = vim.tbl_deep_extend('force', config.modules, user_data)
+
+  config.ensure_installed = user_data.ensure_installed
+  require'nvim-treesitter.install'.ensure_installed(config.ensure_installed)
```

Comment:
We could also just not store `ensure_installed`. We don't need it any more. Actually, I should also delete `enure_installed` from `config.modules`.